### PR TITLE
Unittest

### PR DESCRIPTION
**Pull request by tomkelele**
11/05/2017 :

### CODE REVIEW

File: .php_cs.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...p\\\\Http\\Controllers\\Auth\\ForgotPasswordController.php":190077705,"app\\\\Http\\Controllers\\Auth\\LoginController.php":2028970787,"app\\\\Http\\Controllers\\Auth\\RegisterController.php":1640983441,"app\\\\Http\\Controllers\\Auth\\ResetPasswordController.php":2034098647,"app\\\\Http\\Controllers\\Controller.php":-805096243,"app\\\\Http\\Controllers\\HomeController.php":1773604563,"app\\\\Http\\Controllers\\MembersController.php":1827520256,"app\\\\Http\\Kernel.php":-1330848431,"app\\\\Http\\Middleware\\EncryptCookies.php":-1506495676,"app\\\\Http\\Middleware\\RedirectIfAuthenticated.php":-1037668089,"app\\\\Http\\Middleware\\TrimStrings.php":-2023380469,"app\\\\Http\\Middleware\\VerifyCsrfToken.php":1431329565,"app\\\\Http\\Requests\\MemBerFormRequest.php":-1186781680,"app\\\\Model\\Member.php":-1713450443,"app\\\\Model\\User.php":-757200516,"app\\\\Providers\\AppServiceProvider.php":-1456856802,"app\\\\Providers\\AuthServiceProvider.php":-1175915757,"app\\\\Providers\\BroadcastServiceProvider.php":-100941022,"app\\\\Providers\\EventServiceProvider.php":-2019622043,"app\\\\Providers\\RouteServiceProvider.php":1655606749,"app\\Http\\Requests\\MemBerFormRequest.php":1532557667}}
```
Comment (4c0n) [inline .php_cs.cache]: This file should not be committed, please remove from git and add to .gitignore.

File: app/Http/Controllers/MembersController.php [hunk 2]
```diff
@@ -4,29 +4,29 @@
 
 use Illuminate\Http\Request;
 use App\Model\Member;
-use App\Http\Requests\MemberFormRequest;
+use App\Http\Requests\MemBerFormRequest;
 
 class MembersController extends Controller
 {
     protected $member;
```
Comment (4c0n) [inline app/Http/Controllers/MembersController.php]: This OK, but you can make it private as well. Try to set it to private by default and change to protected if you actually need it to be protected.

File: app/Model/Member.php [hunk 3]
```diff
@@ -7,45 +7,45 @@
...
             $imageName = 'default.png';
         }
-    	$member->photo = $imageName;
-    	$member->save();
+        $member->photo = $imageName;
+        $member->save();
     }
 
-    public function getDelete ($id)
+    public function getDelete($id)
```
Comment (4c0n) [inline app/Model/Member.php]: I think delete() would be a better name. Methods that start with get are called accessor methods and are supposed to return something.

File: public/js/lib/controller/MemberController.js [hunk 4]
```diff
@@ -69,32 +69,34 @@ app.controller('MemberController', function ($scope, $http) {
...
+					transformRequest: function (data, headersGetter) {
+						var formData = new FormData();
+						angular.forEach(data, function (value, key) {
+							formData.append(key, value);
+						});
+						return formData;
+					}
+				}).then(function successCallback(response) {
+					$scope.members = response.data;
+					$('#myModal').modal('hide');
```
Comment (4c0n) [inline public/js/lib/controller/MemberController.js]: I think you should be able to do this without jQuery. Mixing Angular and jQuery is generally considered bad practice.

File: tests/Feature/MembersControllerTest.php [hunk 5]
```diff
@@ -10,41 +10,103 @@
...
-	use DatabaseMigrations;
-	use WithoutMiddleware;
+    use DatabaseMigrations;
     /**
      * A basic test example.
      *
      * @return void
      */
-    public function testExample()
+    public function testLoginDashboard()
```
Comment (4c0n) [inline tests/Feature/MembersControllerTest.php]: What about a negative test case? If there is no user is something else supposed to happen?

File: tests/Feature/MembersControllerTest.php [hunk 6]
```diff
@@ -10,41 +10,103 @@
...
+            'address' => $dataEdit['address'],
+            'age' => $dataEdit['age'],
+            'gender' => $dataEdit['gender'],
+        ]);
+    }
+
+    public function testGetDetailMember()
+    {
+        $user = factory(\App\Model\User::class)->create();
+        $member = factory(\App\Model\Member::class)->create([
```
Comment (4c0n) [inline tests/Feature/MembersControllerTest.php]: You add this member in more than one test. Maybe you can use a setUp() method to add it, or separate the member creation in a separate method.

File: tests/Feature/ValidationWhenAddNewMemberTest.php [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+        $this->actingAs($user)->call('POST', 'dashboard/member/insert', $data);
+        $this->assertDatabaseHas('members', [
+            'name' => $data['name'],
+            'address' => $data['address'],
+            'age' => $data['age'],
+            'gender' => $data['gender'],
+        ]);
+    }
+
+    public function testAddNewMemberWith301CharactersOnAddress()
```
Comment (4c0n) [inline tests/Feature/ValidationWhenAddNewMemberTest.php]: Please also assert the responses of the requests. I'm sure you are using a JSON response and response code.

File: .php_cs.cache (hunk 1 above)
Comment (tomkelele) [inline .php_cs.cache]: Ok I got it.

File: app/Model/Member.php (hunk 3 above)
Comment (tomkelele) [inline app/Model/Member.php]: Yeah sure i will change it.

File: tests/Feature/ValidationWhenAddNewMemberTest.php (hunk 7 above)
Comment (tomkelele) [inline tests/Feature/ValidationWhenAddNewMemberTest.php]: Ok i will do it today.

File: tests/Feature/MembersControllerTest.php (hunk 5 above)
Comment (tomkelele) [inline tests/Feature/MembersControllerTest.php]: I will add it today. Thanks for remind me.

**Pull request merged**